ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/composables/TierSwitcher.kt

```diff
@@ -0,0 +1,132 @@
+package com.revenuecat.purchases.ui.revenuecatui.composables
+
+import androidx.compose.animation.core.animateDpAsState
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.onSizeChanged
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.TemplateConfiguration
+
+private object TierSwitcherUIConstants {
+    const val roundedCorner = 50
+    val height = 40.dp
+    val selectedTierPadding = 2.dp
+}
+
+@Composable
+@Suppress("LongParameterList")
+internal fun SelectedTierView(
+    selectedTier: TemplateConfiguration.TierInfo,
+    backgroundSelectedColor: Color,
+    foregroundSelectedColor: Color,
```

Comment:
There are a lot of places in this PR where there's a background and a foreground color being passed around. Maybe we could create a data class that holds a foreground and a background color? I think that will prevent issues, with passing one instead of another to functions, and also reduce parameters in a bunch of functions